### ISSUE TITLE
No way to configure HEC HttpClient to use proxy

### ISSUE THREAD

**Issue by andy-paine** [splunk/splunk-library-javalogging#117]
Version of this project you're using: 1.6.2
Platform version: Debian Stretch
Framework version: JRE 11
Splunk version: 7.0.8.5
Other relevant information:
We are using the `HttpEventCollectorLogbackAppender` to send some of our logs to splunkcloud. The app sending these logs lives behind an egress proxy (for allowing only certain domains etc.)
The `HttpEventCollectorSender` class news up an Apache `HttpClient` using the `HttpAsyncClients.custom()` method. Unfortunately the Apache `HttpClient` class does not respect the `https.proxyHost`/`http.proxyPort` system properties. As there is no way for us to new up the client ourselves and there are no proxy settings that we can pass in to the `HttpEventCollectorLogbackAppender`, we cannot get the appender to reach splunkcloud through the egress proxy.

It would be great if either the `HttpClient` could use the system properties (`HttpAsyncClients.custom().useSystemProperties()`) or if the proxy settings could be passed into the `HttpEventCollectorLogbackAppender` constructor so that we can correctly configure the client.

**Comment by shakeelmohamed** [comment]
Hi @andy-paine,

Thanks for creating an issue. We're open to a pull request for making the changes to support your use case, I think this is a reasonable change to bring in.

**Comment by fudcom** [comment]
I agree with this one, we needed also need the useSystemProperties(). A protected method would have been nice to at least override or enrich the HTTPAsyncClients settings.

**Comment by rmozone** [comment]
Hello! Any news regarding this issue?
